engine/crates/validation/src/validate/input_object_cycles.rs

```diff
@@ -26,14 +30,20 @@ fn references_input_object_rec<'a>(
         }
 
         let field_type_name = super::extract_type_name(&field.ty.node.base);
+
         if field_type_name == name {
             return Some(vec![field.name.node.as_str()]);
         }
 
+        if visited.contains(field_type_name) {
```

Comment:
So the error was already prepared but you had an infinite loop? funny^^

Reply:
yes :/